src/main/java/io/vertx/ext/amqp/AmqpConnection.java

```diff
@@ -79,19 +79,47 @@
    * @return the connection.
    */
   @Fluent
-  AmqpConnection receiver(String address, AmqpReceiverOptions receiverOptions, Handler<AmqpMessage> messageHandler,
-                          Handler<AsyncResult<AmqpReceiver>> completionHandler);
+  AmqpConnection createReceiver(String address, AmqpReceiverOptions receiverOptions,
+    Handler<AmqpMessage> messageHandler,
+    Handler<AsyncResult<AmqpReceiver>> completionHandler);
+
+  /**
+   * Creates a dynamic receiver. The address is provided by the broker and is available in the {@code completionHandler},
+   * using the {@link AmqpReceiver#address()} method. this method is useful for request-reply to generate a unique
+   * reply address.
+   *
+   * @param completionHandler the completion handler, called when the receiver has been created and opened.
+   * @return the connection.
+   */
+  @Fluent
+  AmqpConnection createDynamicReceiver(Handler<AsyncResult<AmqpReceiver>> completionHandler);
 
   /**
-   * Creates a sender used to send messages to the given address. If no address (i.e null) is specified then a
-   * sender will be established to the 'anonymous relay' and each message must specify its destination address.
+   * Creates a sender used to send messages to the given address. The address must be set. For anonymous sender, check
+   * {@link #createAnonymousSender(Handler)}.
    *
-   * @param address           The target address to attach to, or {@code null} to attach to the anonymous relay.
+   * @param address           The target address to attach to, must not be {@code null}
    * @param completionHandler The handler called with the sender, once opened
    * @return the connection.
+   * @see #createAnonymousSender(Handler)
+   */
+  @Fluent
+  AmqpConnection createSender(String address, Handler<AsyncResult<AmqpSender>> completionHandler);
+
+  /**
+   * Creates an anonymous sender.
+   *
+   * The sender will be established to the 'anonymous relay' and each message must specify its destination address.
+   *
+   * Unlike "regular" sender, this sender is not associated to a specific address, and each message sent must provide
+   * an address. This method is generally used in request-reply scenarios where you need a sender to send the reply,
```

Comment:
I would remove 'This method is generally used...', as I dont think its entirely accurate or necessarily needed. Its common to use them for non request-response cases, or use fixed addresses for request-response.